classes/firmware-qcom-sign.bbclass

```diff
@@ -0,0 +1,133 @@
+sign_verify_file () {
+	file="$(basename $1)"
+	filedir="$(dirname "$1")"
+	sectools="${QCOM_FIRMWARE_SIGN_SECTOOLS_DIR}/sectools"
+	secprofile="${QCOM_FIRMWARE_SIGN_SECPROFILE}"
+	root_hash="0x$(cat ${QCOM_FIRMWARE_SIGN_KEY_DIR}/sha384_roots_hash.txt | cut -d' ' -f2)"
+
+	newline="\n"
+
+	image_id_mapping="\
+		a660_zap.mbn GPU-MICRO-CODE 1 ${newline}\
+		adsp.mbn ADSP 1 ${newline}\
+		aop.mbn AOP 0 ${newline}\
+		cdsp.mbn CDSP 1 ${newline}\
+		cpucp.elf CPUCP 0 ${newline}\
+		devcfg.mbn TZ-DEVCFG 0 ${newline}\
+		hypvm.mbn QHEE 0 ${newline}\
+		imagefv.elf UEFIFV 0 ${newline}\
+		ipa_fws.mbn IPA-FW 1 ${newline}\
+		loadalgota64.mbn TZ-APP-OEM 1 ${newline}\
+		msbtfw11.mbn SKIP 0 ${newline}\
+		multi_image.mbn OEM-MISC 0 ${newline}\
+		prog_firehose_ddr.elf DEVICE-PROGRAMMER 0 ${newline}\
+		prog_firehose_lite.elf DEVICE-PROGRAMMER 0 ${newline}\
+		qupv3fw.elf QUPV3 0 ${newline}\
+		sec.elf SEC-ELF 0 ${newline}\
+		shrm.elf SHRM 0 ${newline}\
+		tz.mbn TZ 0 ${newline}\
+		uefi.elf UEFI 0 ${newline}\
+		uefi_sec.mbn TZ-APP-OEM 0 ${newline}\
+		vpu20_1v.mbn VENUS-FW 0 ${newline}\
+		vpu30_4v.mbn VENUS-FW 0 ${newline}\
+		vpu30_4v_16mb.mbn VENUS-FW 0 ${newline}\
+		wpss.mbn WPSS 1 ${newline}\
+		xbl_config.elf XBL-CONFIG 0 ${newline}\
+		xbl_config_gunyah.elf XBL-CONFIG 0 ${newline}\
+		xbl_config_kvm.elf XBL-CONFIG 0 ${newline}\
+		xbl.elf XBL 0 ${newline}\
+		XblRamdump.elf XBL-RAM-DUMP 0 ${newline}\
+		DigestsToSign.bin.mbn VIP 0 ${newline}\
+		FD02C9DA-306C-48C7-A49C-BBD827AE86EE.mbn TZ-APP-OEM 0 ${newline}\
```

Comment:
This list is rb3gen2/kodiak specific, no? Just thinking as we add support for signing for other targets such as lemans.